lib/shared/runtime-info.js

```diff
@@ -162,6 +162,7 @@ function version() {
 //------------------------------------------------------------------------------
 
 module.exports = {
+    __esModule: true,
```

Comment:
1. First I refactored to this syntax, but `proxquire` does not play well with it (so tests were failing):

```js
{ environment, version } = require("./shared/runtime-info"),
```

So I had no choice but to revert that change to use the "default import":

```js
RuntimeInfo = require("./shared/runtime-info"),
```

That causes ambiguity in terms of intention, so I've added the property. I've explained more about this here:

- https://github.com/eslint/eslint/pull/18005#discussion_r1458414116
- https://knip.dev/guides/commonjs

The name of the property could be anything, I think `__esModule` is common (in bundled code) and describes intent well, but if we think it might cause issues we can rename it.

Reply:
Gotcha, makes sense. Can you add a comment above explaining that this property makes Knip treat the default export correctly? I could see people getting confused and perhaps unintentionally removing it in the future.